### PR TITLE
Fix grpc query failure on legacy blocks

### PR DESCRIPTION
**Pull request by yihuang**
## Description
Mainly `BaseFee` and `EthCall`.
Solution:
- since the grpc query handlers are used for all versions of the blocks, it needs to be compatible with legacy formats.

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Review by fedekunze — COMMENTED** [review]
thanks @yihuang. Can you add unit and integration tests for this change?

**Comment by yihuang** [comment]
> thanks @yihuang. Can you add unit and integration tests for this change?

sure, I'm still investigating another case with `eth_call`, probably similar root cause.

File: CHANGELOG.md [hunk 1]
```diff
@@ -66,6 +66,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (rpc) [#1179](https://github.com/evmos/ethermint/pull/1179) Fix gas used in traceTransaction response.
 * (rpc) [#1284](https://github.com/evmos/ethermint/pull/1284) Fix internal trace response upon incomplete `eth_sendTransaction` call.
 * (rpc) [#1340](https://github.com/evmos/ethermint/pull/1340) Fix error response when `eth_estimateGas` height provided is not found.
+* (rpc) [#]() Make BaseFee grpc query able to query v1 state.
```
Comment (yihuang) [inline CHANGELOG.md:69]:
```suggestion
* (rpc) [#1354](https://github.com/evmos/ethermint/pull/1354) Make BaseFee grpc query able to query v1 state.
```

**Review by fedekunze — APPROVED** [review]

File: tests/integration_tests/configs/upgrade-test-package.nix [hunk 2]
```diff
@@ -0,0 +1,16 @@
+let
+  pkgs = import ../../../nix { };
+  fetchEthermint = rev: builtins.fetchTarball "https://github.com/evmos/ethermint/archive/${rev}.tar.gz";
+  released = pkgs.buildGo118Module rec {
+    name = "ethermintd";
+    src = fetchEthermint "f21592ebfe74da7590eb42ed926dae970b2a9a3f";
```
Comment (yihuang) [inline tests/integration_tests/configs/upgrade-test-package.nix:6]:
```suggestion
    # the commit before https://github.com/evmos/ethermint/pull/943
    src = fetchEthermint "f21592ebfe74da7590eb42ed926dae970b2a9a3f";
```

**Comment by fedekunze** [comment]
@yihuang some unit tests are failing

**Comment by yihuang** [comment]
interesting, it doesn't reproduce locally, can you try rerun?

**Comment by fedekunze** [comment]
```
--- FAIL: TestKeyring (0.29s)
    algorithm_test.go:78: 
        	Error Trace:	/home/runner/work/ethermint/ethermint/crypto/hd/algorithm_test.go:78
        	Error:      	Not equal: 
        	            	expected: "0x25635F7750082d85CcA2D97d1e6086490cBeD0D0"
        	            	actual  : "0x1738E2725c0122705265421F8173AfA75c004CB6"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-0x25635F7750082d85CcA2D97d1e6086490cBeD0D0
        	            	+0x1738E2725c0122705265421F8173AfA75c004CB6
        	Test:       	TestKeyring
```

**Comment by yihuang** [comment]
> ```
> --- FAIL: TestKeyring (0.29s)
>     algorithm_test.go:78: 
>         	Error Trace:	/home/runner/work/ethermint/ethermint/crypto/hd/algorithm_test.go:78
>         	Error:      	Not equal: 
>         	            	expected: "0x25635F7750082d85CcA2D97d1e6086490cBeD0D0"
>         	            	actual  : "0x1738E2725c0122705265421F8173AfA75c004CB6"
>         	            	
>         	            	Diff:
>         	            	--- Expected
>         	            	+++ Actual
>         	            	@@ -1 +1 @@
>         	            	-0x25635F7750082d85CcA2D97d1e6086490cBeD0D0
>         	            	+0x1738E2725c0122705265421F8173AfA75c004CB6
>         	Test:       	TestKeyring
> ```

It don't reproduce when run locally, and didn't seem to be related to this PR, maybe has sth with SDK update.

**Pull request merged**